docs/src/main/paradox/java/http/migration-guide/migration-within-10.0.x-Api-May-Change.md

```diff
@@ -0,0 +1,14 @@
+# Migration Guide within Akka HTTP 10.0.x
+
+## General Notes
+Akka HTTP is binary backwards compatible within for all version within the 10.0.x range. However, there are a set of APIs
+that are marked with the special annotation `@ApiMayChange` where this binary backwards compatibility is not guaranteed.
+See @extref:[The @DoNotInherit and @ApiMayChange markers](akka-docs:common/binary-compatibility-rules.html#The_@DoNotInherit_and_@ApiMayChange_markers) in the Akka documentation for further information.
```

Comment:
~~in the Akka documentation~~

We are in the docs already so no need to explain that it'll be in the docs